### PR TITLE
Generate route with prefix

### PR DESCRIPTION
**Pull request by anqiansong**
* Supported route generates with prefix

### CODE REVIEW

File: tools/goctl/api/tsgen/gen.go [hunk 1]
```diff
@@ -32,6 +34,14 @@ func TsCommand(c *cli.Context) error {
 		return err
 	}
 
+	service := api.Service
+	for _, g := range service.Groups {
```
Comment (kevwan) [inline tools/goctl/api/tsgen/gen.go]: duplicated code?

File: tools/goctl/util/string.go [hunk 2]
```diff
@@ -110,3 +110,8 @@ func isGolangKeyword(s string) bool {
 	_, ok := goKeyword[s]
 	return ok
 }
+
+func TrimSpace(s string) string {
```
Comment (kevwan) [inline tools/goctl/util/string.go:114]: strings.TrimSpace?

**Review by kevwan — CHANGES_REQUESTED** [review]

File: tools/goctl/api/tsgen/gen.go (hunk 1 above)
Comment (anqiansong) [reply tools/goctl/api/tsgen/gen.go]: see spec.Service.JoinPrefix

File: tools/goctl/util/string.go (hunk 2 above)
Comment (anqiansong) [reply tools/goctl/util/string.go]: Done.

File: tools/goctl/api/ktgen/cmd.go [hunk 3]
```diff
@@ -27,6 +27,8 @@ func KtCommand(c *cli.Context) error {
 		return e
 	}
 
+	service := api.Service.JoinPrefix()
+	api.Service = service
```
Comment (kevwan) [inline tools/goctl/api/ktgen/cmd.go:31]: api.Service = api.Service.JoinPrefix()

File: tools/goctl/api/dartgen/gen.go [hunk 4]
```diff
@@ -25,6 +25,7 @@ func DartCommand(c *cli.Context) error {
 		return err
 	}
 
+	api.Service= api.Service.JoinPrefix()
```
Comment (kevwan) [inline tools/goctl/api/dartgen/gen.go:28]: format code

File: tools/goctl/api/javagen/gen.go [hunk 5]
```diff
@@ -28,6 +28,7 @@ func JavaCommand(c *cli.Context) error {
 		return err
 	}
 
+	api.Service= api.Service.JoinPrefix()
```
Comment (kevwan) [inline tools/goctl/api/javagen/gen.go:31]: format

**Review by kevwan — CHANGES_REQUESTED** [review]

**Review by kevwan — APPROVED** [review]

**Pull request merged**